sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/SymbolicExecutionAnalyzerFactoryTest.cs

```diff
@@ -0,0 +1,109 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2020 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Rules.SymbolicExecution;
+
+namespace SonarAnalyzer.UnitTest.Rules.SymbolicExecution
+{
+    [TestClass]
+    public class SymbolicExecutionAnalyzerFactoryTest
+    {
+        [TestMethod]
+        public void SupportedDiagnostics_ReturnsSymbolicExecutionRuleDescriptors()
+        {
+            // Currently only EmptyNullableValueAccess was updated to use the new analyzer.
+
+            var sut = new SymbolicExecutionAnalyzerFactory();
+            var supportedDiagnostics = sut.SupportedDiagnostics.Select(descriptor => descriptor.Id).ToList();
+
+            CollectionAssert.AreEquivalent(supportedDiagnostics, new[] {"S3655", "S3966"});
```

Comment:
maybe store in class constants the rule ids (S3655) with the name of the actual rule? it would make the tests more readable